Lib/test/test_tools/test_i18n.py

```diff
@@ -72,3 +73,20 @@ def test_POT_Creation_Date(self):
 
             # This will raise if the date format does not exactly match.
             datetime.strptime(creationDate, '%Y-%m-%d %H:%M%z')
+
+    def test_files_list(self):
+        """Make sure the directories are inspected for source files
+           bpo-31920
+        """
+        source_dir = 'pypkg'
+        text = 'Text to translate'
+        with temp_cwd(None) as cwd:
+            with temp_dir(os.path.join(cwd, source_dir)) as sdir:
+                with open(os.path.join(sdir, 'pymod.py'), 'w') as sfile:
```

Comment:
Would be nice to put the test file deeper in the directory tree, e.g. in `os.path.join(sdir, 'pypkg', 'pymod.py')`. This would test that directories are searched recursively.